service/worker/esanalyzer/workflow.go

```diff
@@ -0,0 +1,389 @@
+// Copyright (c) 2021 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package esanalyzer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/activity"
+	cclient "go.uber.org/cadence/client"
+	"go.uber.org/cadence/workflow"
+
+	"github.com/uber/cadence/common/log/tag"
+	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/types"
+)
+
+type (
+	contextKey string
+
+	Duration struct {
+		AvgExecTime float64 `json:"value"`
+	}
+
+	WorkflowTypeInfo struct {
+		Name         string   `json:"key"`
+		NumWorfklows int64    `json:"doc_count"`
+		Duration     Duration `json:"duration"`
+	}
+
+	WorkflowInfo struct {
+		DomainID   string
+		WorkflowID string
+		RunID      string
+	}
```

Comment:
aah, I see.  in that case it may be fine.
personally I've gotten a LOT of good results out of explicitly marking every serializable thing (and adding a linter based on that).  in literally every repo that I've applied that pattern, I've found things that were unintentionally being serialized, and were causing subtle bugs :|